src/ott/solvers/linear/discrete_barycenter.py

```diff
@@ -33,67 +34,87 @@ class SinkhornBarycenterOutput(NamedTuple):  # noqa: D101
+    inner_iterations: number of iterations run before recomputing errors.
+    min_iterations: number of iterations run without checking whether
+      termination criterion is true.
+    max_iterations: maximal number of iterations.
+    lse_mode: sets computations in kernel (``False``) or log-sum-exp mode.
+    debiased: uses debiasing correction to avoid blur due to entropic
+      regularization.
   """
-  batch_size, num_a = a.shape
-  _, num_b = geom.shape
-
-  if weights is None:
-    weights = jnp.ones((batch_size,)) / batch_size
-  if weights.shape[0] != batch_size:
-    raise ValueError(f'weights must have size `{batch_size}`.')
-
-  if dual_initialization is None:
-    # initialization strategy from https://arxiv.org/pdf/1503.02533.pdf, (3.6)
-    dual_initialization = geom.apply_cost(a.T, axis=0).T
-    dual_initialization -= jnp.average(
-        dual_initialization, weights=weights, axis=0
-    )[jnp.newaxis, :]
-
-  if debiased and not geom.is_symmetric:
-    raise ValueError('Geometry must be symmetric to use debiased option.')
-  norm_error = (norm_error,)
-  return _discrete_barycenter(
-      geom, a, weights, dual_initialization, threshold, norm_error,
-      inner_iterations, min_iterations, max_iterations, lse_mode, debiased,
-      num_a, num_b
-  )
+
+  def __init__(
+      self,
+      threshold: float = 1e-2,
+      norm_error: int = 1,
+      inner_iterations: float = 10,
+      min_iterations: int = 0,
+      max_iterations: int = 2000,
+      lse_mode: bool = True,
+      debiased: bool = False
+  ):
+    self.threshold = threshold
+    self.norm_error = norm_error
+    self.inner_iterations = inner_iterations
+    self.min_iterations = min_iterations
+    self.max_iterations = max_iterations
+    self.lse_mode = lse_mode
+    self.debiased = debiased
+
+  def __call__(
+      self,
+      fixed_bp: barycenter_problem.FixedBarycenterProblem,
+      dual_initialization: Optional[jnp.ndarray] = None,
+  ) -> SinkhornBarycenterOutput:
+    """Solve barycenter problem, possibly using clever initialization."""
+    geom = fixed_bp.geom
+    a = fixed_bp.a
+    num_a, num_b = geom.shape
+
+    weights = fixed_bp.weights
+
+    if dual_initialization is None:
+      # initialization strategy from :cite:`cuturi:15`, (3.6).
+      dual_initialization = geom.apply_cost(a.T, axis=0).T
+      dual_initialization -= jnp.average(
+          dual_initialization, weights=weights, axis=0
+      )[jnp.newaxis, :]
+
+    if self.debiased and not geom.is_symmetric:
+      raise ValueError('Geometry must be symmetric to use debiased option.')
+    norm_error = (self.norm_error,)
+    return _discrete_barycenter(
+        geom, a, weights, dual_initialization, self.threshold, norm_error,
+        self.inner_iterations, self.min_iterations, self.max_iterations,
+        self.lse_mode, self.debiased, num_a, num_b
+    )
+
+  def tree_flatten(self):  # noqa: D102
+    aux = vars(self).copy()
+    aux.pop('threshold')
+    return [
```

Comment:
Can't this be in one line or does `yapf` object?

Reply:
indeed, `yapf` makes the change... tried to revert to one line but pre-commit changes